services/provider/server/server.go

```diff
@@ -278,23 +278,29 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
 	}
 
-	// Service for monitoring endpoints
-	scMonitoring := &v1.Service{}
-	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	// Get mgr pod hostIP
+	podList := &corev1.PodList{}
+	err = s.client.List(ctx, podList, client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}))
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
 	}
 
-	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
-		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	var port int32
+	for i := range podList.Items[0].Spec.Containers[0].Ports {
```

Comment:
Why do we assume the pod has containers? what if Ceph will add additional containers? do we assume it will always be the first one?

I think we should search for the container by name.

Reply:
even if we have more than one container we should be able to use any of them IMO.

Reply:
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    openshift.io/scc: rook-ceph
    prometheus.io/port: "9283"
    prometheus.io/scrape: "true"
  creationTimestamp: "2022-02-17T09:26:19Z"
  generateName: rook-ceph-mgr-a-5544b8b8d5-
  labels:
    app: rook-ceph-mgr
    app.kubernetes.io/component: cephclusters.ceph.rook.io
    app.kubernetes.io/created-by: rook-ceph-operator
    app.kubernetes.io/instance: a
    app.kubernetes.io/managed-by: rook-ceph-operator
    app.kubernetes.io/name: ceph-mgr
    app.kubernetes.io/part-of: ocs-storagecluster-cephcluster
    ceph_daemon_id: a
    ceph_daemon_type: mgr
    instance: a
    mgr: a
    pod-template-hash: 5544b8b8d5
    rook.io/operator-namespace: openshift-storage
    rook_cluster: openshift-storage
  name: rook-ceph-mgr-a-5544b8b8d5-5czgz
  namespace: openshift-storage
```

If we have more than one pod they should be like `rook-ceph-mgr-b`, ` rook-ceph-mgr-c` and we should be able to use any of them.

Reply:
@iamniting I think you are talking about multiple pods, not containers. 
If Ceph will add a sidecar container to the pod and have it as the first container our code will break. 
We should select the right container by name

Reply:
my bad, make sense.